server/src/test/java/org/elasticsearch/index/mapper/blockloader/TextFieldWithParentBlockLoaderTests.java

```diff
@@ -9,51 +9,108 @@
 
 package org.elasticsearch.index.mapper.blockloader;
 
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.elasticsearch.datageneration.DocumentGenerator;
 import org.elasticsearch.datageneration.FieldType;
+import org.elasticsearch.datageneration.MappingGenerator;
+import org.elasticsearch.datageneration.Template;
 import org.elasticsearch.datageneration.datasource.DataSourceHandler;
 import org.elasticsearch.datageneration.datasource.DataSourceRequest;
 import org.elasticsearch.datageneration.datasource.DataSourceResponse;
-import org.elasticsearch.datageneration.datasource.DefaultMappingParametersHandler;
 import org.elasticsearch.index.mapper.BlockLoaderTestCase;
+import org.elasticsearch.index.mapper.BlockLoaderTestRunner;
+import org.elasticsearch.index.mapper.MapperServiceTestCase;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
 
-import java.util.HashMap;
+import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 
-public class TextFieldWithParentBlockLoaderTests extends BlockLoaderTestCase {
-    public TextFieldWithParentBlockLoaderTests(Params params) {
-        // keyword because we need a keyword parent field
-        super(FieldType.KEYWORD.toString(), List.of(new DataSourceHandler() {
+import static org.elasticsearch.index.mapper.BlockLoaderTestCase.buildSpecification;
+import static org.elasticsearch.index.mapper.BlockLoaderTestCase.hasDocValues;
+
+public class TextFieldWithParentBlockLoaderTests extends MapperServiceTestCase {
```

Comment:
This is now a dedicated test for this case and it does not include "normal" tests that are covered in `TextFieldBlockLoaderTests`.